usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -1036,6 +1036,8 @@ uint64_t l2arc_feed_min_ms = L2ARC_FEED_MIN_MS;	/* min interval milliseconds */
 boolean_t l2arc_noprefetch = B_TRUE;		/* don't cache prefetch bufs */
 boolean_t l2arc_feed_again = B_TRUE;		/* turbo warmup */
 boolean_t l2arc_norw = B_TRUE;			/* no reads during writes */
+boolean_t arc_evict_l2_first = B_FALSE;		/* first evict buffers from ARC which are in L2ARC */
+boolean_t arc_evict_l2_only = B_FALSE;		/* only evict buffers from ARC which are in L2ARC */
```

Comment:
`arc_evict_l2_first` will first choose L2ARC bufs. If it did not found enough of them to reach the requested amount (`total_evicted < bytes`), it will catch non-L2ARC bufs.

`arc_evict_l2_only` will only choose L2ARC bufs. As we could run out of memory in such a scenario, it will catch non-L2ARC bufs if it did not found any L2ARC bufs to evict (see line 3086, the `total_evicted > 0` condition).

Of course `arc_evict_l2_only` feeds L2ARC better than `arc_evict_l2_first`.
Drawback of `arc_evict_l2_only` is that it could lead to ZFS throttle waiting for some ARC free space, so ones using it may have to tune `l2arc_headroom` and `l2arc_feed_min_ms` so that L2ARC is fed fast enough.